packages/db/test/user-storage.spec.js

```diff
@@ -0,0 +1,201 @@
+      dagSize: uploadSize
+    })
+    await createUpload(dbClient, Number(user3._id), Number(authKey3), cids[1], {
+      dagSize: Math.round(uploadSize * 1.2)
+    })
+    await createPsaPinRequest(dbClient, authKey3, cidsPinned[0], {
+      dagSize: pinnedSize,
+      pins
+    })
+
+    user4 = await createUser(dbClient, {
+      name: 'test4-name',
+      email: 'test4@email.com'
+    })
+    authKey4 = await createUserAuthKey(dbClient, Number(user4._id), {
+      name: 'test4-key'
+    })
+    await createUserTag(dbClient, Number(user4._id), {
+      tag: 'StorageLimitBytes',
+      value: '2199023255552'
+    })
+    const cid = 'bafybeibvuy3vcepqxy4plr34twv22vvxol2jjhmjxcrcvuhea5226whpsm'
+    await createUpload(dbClient, Number(user4._id), Number(authKey4), cid, {
+      dagSize: largeFileSize
+    })
+
+    // User               | upload  | pinned  | quota   | percentage of quota
+    // ----------------------------------------------------------------------
+    // test1@email.com    | 1       | 2       | 1TiB    | < 75%
+    // test2@email.com    | 2       | 0       | 1TiB    | 80%
+    // test3@email.com    | 2       | 1       | 1TiB    | 90%
+    // test4@email.com    | 1 (XL)  | 0       | 2TiB    | > 90%
+  })
+
+  it('returns user details needed for email', async () => {
+    const users = await dbClient.getUsersByStorageUsed({
+      fromPercent: 50
+    })
+
+    assert.strictEqual(users.length, 4, 'All users returned')
+    assert(users[0].name, 'User has a name')
+    assert(users[0].email, 'User has an email')
+    assert(users[0].storageQuota, 'User has storage quota')
+    assert(users[0].storageUsed, 'User has storage quota used')
+  })
+
+  it('retrieves correct users in a range', async () => {
+    const users = await dbClient.getUsersByStorageUsed({
+      fromPercent: 70,
+      toPercent: 95
+    })
+    assert.strictEqual(users.length, 2, 'Users with quota in a range')
+    assert.strictEqual(users[0].email, 'test3@email.com')
+    assert.strictEqual(users[1].email, 'test2@email.com')
+  })
+
+  it('retrieves users sorted by used storage descending', async () => {
+    const users = await dbClient.getUsersByStorageUsed({
+      fromPercent: 50,
+      toPercent: 95
+    })
+    assert.strictEqual(users.length, 3, 'Users with quota in a range')
+    assert.strictEqual(users[0].email, 'test3@email.com', 'Sorted by used storage descending')
+    assert.strictEqual(users[1].email, 'test2@email.com')
+    assert.strictEqual(users[2].email, 'test1@email.com')
+  })
+
+  it('uses user quota instead of default if one has been specified', async () => {
+    const users = await dbClient.getUsersByStorageUsed({
+      fromPercent: 90
+    })
+    assert.strictEqual(users.length, 2, 'Users with non-default quota included')
+    assert.strictEqual(users[0].email, 'test4@email.com')
+    assert.strictEqual(users[1].email, 'test3@email.com')
+  })
+
+  it('checks email log before sending', async () => {
```

Comment:
Wouldn't it be better to have the email related tests in their own file?

Reply:
Agree, moved to `email.spec.js` in the `db` package.